src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs

```diff
@@ -244,12 +244,10 @@ public override void ExecuteCmdlet()
                             {
                                 long currentBlockSize = offset + blockSize < fileSize ? blockSize : fileSize - offset;
 
-                                // Only need to create new buffer when chunk size change
-                                if (currentBlockSize != lastBlockSize)
-                                {
-                                    buffer = new byte[currentBlockSize];
-                                    lastBlockSize = currentBlockSize;
-                                }
+                                // create new buffer, the old buffer will be GC
+                                buffer = new byte[currentBlockSize];
+                                lastBlockSize = currentBlockSize;
+
                                 await stream.ReadAsync(buffer: buffer, offset: 0, count: (int)currentBlockSize);
```

Comment:
Noticed the code below ignores the number of bytes read by `ReadAsync` and just assumes the buffer is full. While that works now, would it be safer to change that in case `FileStream` changes in the future?

Reply:
@billings7 

Thanks for the review and the comment!
I have modified the code to respect the bytes count read by `ReadAsync`. 
I have tested that works with different file size (from 1KB up to 2GB), with random the bytes count read by `ReadAsync`.